packages/typescript/src/outputFile.ts

```diff
@@ -72,3 +76,36 @@ export default function findTypescriptOutput(
     declarations: emittedFileNames.filter((name) => name !== codeFile && name !== mapFile)
   };
 }
+
+export function normalizePath(fileName: string) {
+  return fileName.split(path.win32.sep).join(path.posix.sep);
+}
+
+export async function emitFile(
+  outputOptions: OutputOptions,
+  outputToFilesystem: boolean | undefined,
+  context: PluginContext,
+  filePath: string,
+  fileSource: string
+) {
+  // TODO: this is confusing, change
+  const fromRollupOutputDirToFile = outputOptions.dir
```

Comment:
This line can probably be combined with the if statement, it only is used in one branch.

```ts
if (!outputOptions.dir) {
  if (outputToFilesystem == null)...
} else {
  const fromRollupOutputDirToFile = path.relative(outputOptions.dir, filePath)
  context.emitFile({ ...
}
```

Reply:
That change results in a failing test:

```
  Error {
    code: 'PLUGIN_ERROR',
    hook: 'generateBundle',
    plugin: 'typescript',
    pluginCode: 'VALIDATION_ERROR',
    message: 'The "fileName" or "name" properties of emitted files must be strings that are neither absolute nor relative paths and do not contain invalid characters, received "../.tsbuildinfo".',
  }
 ```
 
 I'll see if I can refactor that in another way.